core/comment_events.js

```diff
@@ -155,21 +183,36 @@ Blockly.Events.CommentChange.prototype.isNull = function() {
  * @param {boolean} forward True if run forward, false if run backward (undo).
  */
 Blockly.Events.CommentChange.prototype.run = function(forward) {
-  var workspace = this.getEventWorkspace_();
-  var comment = workspace.getCommentById(this.commentId);
+  var comment = this.getComment_();
   if (!comment) {
     console.warn('Can\'t change non-existent comment: ' + this.commentId);
     return;
   }
   var contents = forward ? this.newContents_ : this.oldContents_;
 
-  comment.setContent(contents);
+  if (contents.hasOwnProperty('minimized')) {
+    if (comment instanceof Blockly.ScratchBlockComment) {
+      // TODO remove this check when workspace comments also get a minimized
+      // state
+      comment.setMinimized(contents.minimized);
+    }
+  }
+  if (contents.hasOwnProperty('width') && contents.hasOwnProperty('height')) {
+    comment.setSize(contents.width, contents.height);
+  }
+  if (contents.hasOwnProperty('text')) {
+    comment.setText(contents.text);
+  }
+
+  console.warn('Unrecognized comment change: ' + JSON.stringify(contents) + '; cannot undo/redo');
```

Comment:
This now fires every single time.  Delete this line.

Reply:
Oops! Totally forgot about that. =]